sources/packages/api/src/database/database.service.spec.ts

```diff
@@ -0,0 +1,29 @@
+import "reflect-metadata";
+require("../../env_setup");
+import { Test, TestingModule } from "@nestjs/testing";
+import { getConnection } from "typeorm";
+import { DatabaseModule } from "./database.module";
+import { DatabaseService } from "./database.service";
+
+describe("Database Service", () => {
+  let service: DatabaseService;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [DatabaseModule],
+    }).compile();
+    await module.init();
+    service = module.get<DatabaseService>(DatabaseService);
+  });
+
+  it("should be defined", () => {
+    expect(service).toBeDefined();
+  });
+
+  it("should create service", async () => {
+    const connection = getConnection();
+    const svc = new DatabaseService(connection);
+    expect(svc).toBeDefined();
+    expect(svc.logger()).toBeDefined();
```

Comment:
Is it possible, using the current design, to replace the current logger implementation with a mocked object?
In the future, if our logger is saving to a file or a database, and I want to execute a unit test is some service that uses the logger, is it possible to have a mocked logger to test this service?

Reply:
It is partial test of the concepts, but my idea is to mock it through LoggerService ....

Reply:
Currently Nest logger is automatically disabled in test env